### PR TITLE
[Mellanox] Implementation for component version listing

### PR DESCRIPTION
**Pull request by skr31**
#### Why I did it
Added ability to show summary of all the component versions on the switch and a comparison between compiled versions and actual versions.
The user will be able to run the 'get_component_versions.py' script which outputs the versions table.
The versions table will also be saved in techsupport.

This PR should be merged before https://github.com/sonic-net/sonic-utilities/pull/3264.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
* Added Make files that add a new file to the system that contains all the versions of the components available at compilation (the versions are taken from corresponding .mk files).
* Added a script that outputs a table that summarizes all the component versions (compilation and actual)

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by skr31** [comment]
This PR should be merged before https://github.com/sonic-net/sonic-utilities/pull/3264.

**Review by nazariig — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

File: platform/mellanox/get_component_versions/get_component_versions.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    # The first layer of the map only has one item
+    for key, value in ccm.items():
+        if not ccm[key]:
+            return UNAVAILABLE_PLATFORM_VERSIONS
+
+        versions_map = ccm[key]
+
+    if not versions_map or len(versions_map) == 0:
+        return UNAVAILABLE_PLATFORM_VERSIONS
```
[the remark this answers is not in the document]
Comment (skr31) [reply platform/mellanox/get_component_versions/get_component_versions.py:82]: Fixed

**Review by nazariig — APPROVED** [review]

**Comment by prsunny** [comment]
@qiluo-msft , @maipbui , can you check the semgrep issue in this PR? Not sure if related to https://github.com/sonic-net/sonic-buildimage/pull/18577

File: platform/mellanox/get_component_versions/get_component_versions.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+    platform_versions = {}
+    for component_name, component in versions_map.items():
+        platform_versions[component_name] = component.get_firmware_version()
+
+    return platform_versions
+
+
+def get_current_version(comp):
+    version = subprocess.run(COMMANDS_FOR_ACTUAL[comp][0], shell=True, stdout=subprocess.PIPE)
```
Comment (maipbui) [inline platform/mellanox/get_component_versions/get_component_versions.py:92]: >version = subprocess.run(COMMANDS_FOR_ACTUAL[comp][0], shell=True, stdout=subprocess.PIPE) [](http://example.com/codeflow?start=4&length=90)

Please evaluate whether subprocess usage is secure or potentially vulnerable
- If a potential vulnerability exists, you should consider revising your implementation
- If it is safe (e.g. not using user input directly or use shell=False), then you can bypass the Semgrep check by `# nosemgrep <rule-id>`

Best practice: [https://semgrep.dev/docs/cheat-sheets/python-command-injection/](https://semgrep.dev/docs/cheat-sheets/python-command-injection/)
Bypass Semgrep: [https://semgrep.dev/docs/ignoring-files-folders-code/#ignore-code-through-nosemgrep](https://semgrep.dev/docs/ignoring-files-folders-code/#ignore-code-through-nosemgrep)

**Review by liat-grozovik — DISMISSED** [review]

**Review by nazariig — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**